components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -1703,8 +1703,7 @@ private String constructAssertionConsumerUrl(MessageContext messageContext) {
         //Note: Do not change to construct the assertionConsumerUrl directly using servicePrefix instead of headers.get("HOST").
         //It always gives IP for host which cause invalid assertionConsumerUrl
         String assertionConsumerUrl = servicePrefix.substring(0, servicePrefix.indexOf("/") + 2) +
-                                      headers.get("HOST") + messageContext.getProperty("REST_API_CONTEXT") +
-                                      "/" + messageContext.getProperty("SYNAPSE_REST_API_VERSION") + "/";
+                headers.get("HOST") + messageContext.getProperty("REST_API_CONTEXT") + "/";
 
```

Comment:
Use AppMConstants.HOST constant.
Better to define REST_API_CONTEXT value as a constant too.